### ISSUE TITLE
duplicate --find-links

### ISSUE THREAD

**Issue by tisdall** [jazzband/pip-tools#1018]
I've managed to get 2 `--find-links` in my requirements.txt when I altered the single value in requirements.in . The first is the one before the change and the second is after.

#### Environment Versions

1. OS Type: MacOS 10.13
1. Python version: 3.7.5
1. pip version:  19.3.1
1. pip-tools version: 4.3.0

#### Steps to replicate

1. add `--find-links file:./local_src` in requirements.in
1. `pip-compile requirements.in`
1. change entry in 1 to `--find-links ./local_src`
1. `pip-compile requirements.in`

#### Expected result
(I would actually expect the result to remain a relative path, but that's #204)
```
--find-links /Users/me/myrepo/./local_src
```

#### Actual result

```
--find-links file:./local_src
--find-links /Users/me/myrepo/./local_src
```

**Comment by atugushev** [comment]
@tisdall thanks for the issue!

**Comment by atugushev** [comment]
FTR, should be fixed by passing deep copied finder here: https://github.com/jazzband/pip-tools/blob/67092445a1c47dc453b8cc9753bb9fbf125beaac/piptools/scripts/compile.py#L290

**Comment by atugushev** [comment]
I've prepared a fix #1029. I'd be appreciated any feedback!
